### ISSUE TITLE
Deleting a TMA grid deletes all regions

### ISSUE THREAD

**Issue by SoerenXD** [qupath/qupath#646]
## IMPORTANT!

This section is **only** for bug reports, which **must** follow the template below.

Please post any other questions to the official QuPath forum at https://forum.image.sc/tags/qupath

If in doubt, use [image.sc](https://forum.image.sc/tags/qupath). This helps us keep things organized.

## Before we begin...

Before submitting your bug report, please check the following:

* [x] I've definitely found a bug (it you're not sure, please use [image.sc](https://forum.image.sc/tags/qupath) instead)
* [x] I've checked https://qupath.github.io for a new release that might already have fixed the issue
* [x] I've checked the [Changelog](https://github.com/qupath/qupath/blob/master/CHANGELOG.md) to see if the bug has already been fixed in the next release
* [x] I've checked for existing GitHub issues describing the same problem

## Bug report

**Describe the bug**
If a TMA grid has been assigned using the TMA dearrayer and the TMA grid is removed later, it will remove all regions along with it, regardless of whether they have been placed n the hierachy or not.

**To Reproduce**
1. Open a TMA image
2. Draw some regions in the TMA cores
3. Run the TMA de-arrayer
4. Delete the TMA-grid
5. All manually drawn regions are gone.

**Expected behavior**
The TMA grid should be removed and the other regions should be kept.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Windows
 - QuPath Version 0.2.3

**Additional context**

**Comment by petebankhead** [comment]
Thanks, I can reproduce the behavior - the entire object hierarchy is removed.
Furthermore, `clearTMAGrid()` no longer works from a script.

**Comment by petebankhead** [comment]
This script may be useful as a workaround, but I haven't checked if it has any unwelcome side-effects. The bug will need fixed in a later release.
```groovy
def hierarchy = getCurrentHierarchy()
hierarchy.clearTMAGrid()
hierarchy.tmaGrid = null
fireHierarchyUpdate()
```

**Comment by petebankhead** [comment]
The changes in https://github.com/petebankhead/qupath/commit/deea6c7d71baee558701f9ae1e5c85434c00384c have hopefully resolved the issue, and will be available in v0.3.
Please do reopen this if you find that it hasn't been fixed after all.
